File: internal/coolify/event_test.go
package coolify

import (
	"encoding/json"
	"testing"
)

func TestEvent_Unmarshal(t *testing.T) {
	raw := `{"success":true,"message":"m","event":"deployment_success","application_name":"app","project":"p"}`
	var e Event
	if err := json.Unmarshal([]byte(raw), &e); err != nil {
		t.Fatal(err)
	}
	if e.Event != "deployment_success" || e.ApplicationName != "app" || e.Project != "p" {
		t.Fatalf("%+v", e)
	}
}

File: src/coolify/event.test.ts
import { describe, expect, it } from "vitest";
import type { CoolifyEvent } from "./event.js";

describe("CoolifyEvent JSON", () => {
  it("unmarshals snake_case fields", () => {
    const raw =
      '{"success":true,"message":"m","event":"deployment_success","application_name":"app","project":"p"}';
    const e = JSON.parse(raw) as CoolifyEvent;
    expect(e.event).toBe("deployment_success");
    expect(e.application_name).toBe("app");
    expect(e.project).toBe("p");
  });
});
